pkg/shardddl/optimism/lock.go

```diff
@@ -264,15 +279,25 @@ func (l *Lock) TrySync(info Info, tts []TargetTable) (newDDLs []string, err erro
 
 		cmp, _ = prevTable.Compare(nextTable) // we have checked `err` returned above.
 		if cmp < 0 {
-			// check for add column with different field lengths
-			err = AddDifferentFieldLenColumns(l.ID, ddls[idx], nextTable, newJoined)
-			if err != nil {
+			// check for add column with a smaller field len
```

Comment:
```suggestion
			// check for add column with a different field len
```

🤔

Reply:
This comment means we add a column with a larger field length before, and then we add a field len with a smaller one.
For example, `alter table t add column (b varchar(11))`(synced) and `alter table t add column (b varchar(6))` (current one).

Reply:
where's logic we let "larger" field length pass 🤔

Reply:
here https://github.com/pingcap/dm/pull/1510/files#diff-110e5d790ae9426fcd1453f3e2a61bb5de5440c0e6ce8dd76d72de4fe2d16432R254